nftables_test.go

```diff
@@ -7666,3 +7667,82 @@ func TestNftablesCompat(t *testing.T) {
 		t.Fatalf("compat policy should conflict and err should not be err")
 	}
 }
+
+func TestNftablesDeadlock(t *testing.T) {
+	helperConn := func(t *testing.T, readBufSize, writeBufSize, wantRules int) (error, int) {
```

Comment:
Can you turn this into a table-driven test for consistency with the existing tests in this repository please? See https://go.dev/blog/subtests for more background.

The big difference is that all parameters should be encapsulated in a table entry (send/receive buffer size in this case, I suppose). What you currently have in `helperConn` should be the actual test function body.

Reply:
fixed!